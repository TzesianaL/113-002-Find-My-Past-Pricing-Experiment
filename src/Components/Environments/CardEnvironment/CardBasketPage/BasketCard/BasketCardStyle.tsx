import { TitleStyle } from 'Components/Shared/Title/TitleStyle'
import styled from 'styled-components'
import { FeaturesStyle } from '../../Features/FeaturesStyle'
/**
 * Styled component is used to decorate CardWrapper component.
 * Props are passed from ThemeProvider component.
 */

export const BasketCardStyle = styled.div.attrs({
  className: 'basketCard',
})`
  background-color: white;
  display: flex;
  flex-direction: column;
  justify-self: center;
  width: 100%;
  // padding: 16px 1em 24px 1em;
  align-items: left;
  border: 7px;
  border-color: ${(props) => props.theme.card?.borderColor || 'gray'};
  border-radius: 7px;
  max-width: 500px;
  color: black;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.12);
  // margin: 0em 1em 1em 1em;

  @media (max-width: 768px) {
    max-width: 350px;
  }
  // font-size: 0.8em;
  .cardPriceDetails {
    display: flex;
    flex-direction: row;
    justify-content: center;
    .cardprice {
      text-align: center;
      font-weight: bold;
      font-size: large;
    }
    .periodPlaceholder {
      font-weight: bold;
      padding: 0.1em;
    }
  }
  ${TitleStyle} {
    color: black;
  }
  ${FeaturesStyle} {
    color: black;
    flex-grow: 1;
  }
`

export const BasketCardWrapper = styled.div.attrs({
  className: 'basketCardWrapper',
})`
  display: flex;
  color: ${(props) =>
    props.theme.card?.color || props.theme.colors?.primary || '#000000'};
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1em 0;
  margin: 1em 0;
  max-width: 50em;
`

export const BasketCardRow = styled.div`
  // display: flex;
  width: 500px;
`
