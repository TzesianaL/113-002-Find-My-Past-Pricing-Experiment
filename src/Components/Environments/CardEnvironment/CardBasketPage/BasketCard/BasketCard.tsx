import React, { FC } from 'react'
import Title from 'Components/Shared/Title/Title'
import { AddonData } from 'types'
import Price from 'Components/Shared/Price/Price'
import BasketRow from '../BasketRow/BasketRow'
import { BasketCardStyle, BasketCardRow } from './BasketCardStyle'

/**
 * CardProps contains all appropriate props we should pass to nested components.
 */
export interface BasketCardProps {
  title: string
  subtitle?: string
  cardImage?: string
  price?: number
  speed: string
  upfrontPrice?: number
}

/**
 * BasketCard is used to wrap all components we render in order to display selected Card.
 * ThemeProvider component is used. By using the theme object we provide same styling options to all components.
 */
const BasketCard: FC<BasketCardProps> = ({
  title,
  speed,
  subtitle,
  price,
  upfrontPrice,
}) => {
  return (
    <BasketCardRow>
      <BasketCardStyle style={{ border: '2px solid black' }}>
        <BasketRow text={speed} />
        <BasketRow text={title} />
        <Title subtitle>
          {subtitle !== undefined && (
            <div dangerouslySetInnerHTML={{ __html: subtitle }} />
          )}
          <br />
          <hr />
          <div style={{ display: 'flex', gap: '1em' }}>
            <Price price={price} pricePeriod="month" />
            <Price price={upfrontPrice} pricePeriod="upfront" />
          </div>
        </Title>
      </BasketCardStyle>
    </BasketCardRow>
  )
}

export default BasketCard
