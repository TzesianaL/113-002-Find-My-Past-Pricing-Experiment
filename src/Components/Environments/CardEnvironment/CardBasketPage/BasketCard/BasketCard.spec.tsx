import React from 'react'
import 'jest-styled-components'
import { render } from '@testing-library/react'
import BasketCard from './BasketCard'

describe('<BasketCard />', () => {
  it('should render correctly Basket Card', () => {
    const { asFragment } = render(
      <BasketCard title="title" cardImage="path to image" subtitle="subtitle" />
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
