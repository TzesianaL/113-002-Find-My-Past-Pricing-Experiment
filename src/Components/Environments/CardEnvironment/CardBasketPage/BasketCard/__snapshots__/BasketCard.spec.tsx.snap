// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketCard /> should render correctly Basket Card 1`] = `
<DocumentFragment>
  .c4 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  font-family: BoldVf !important;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c5 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #ffffff;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 img {
  height: 36px;
  padding: 0 0.25em;
}

.c1 {
  background-color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  padding: 16px 1em 24px 1em;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 7px;
  border-color: #000000;
  border-radius: 7px;
  max-width: 100%;
  color: black;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.12);
  margin: 0em 1em 1em 1em;
}

.c1 .cardPriceDetails {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 .cardPriceDetails .cardprice {
  text-align: center;
  font-weight: bold;
  font-size: large;
}

.c1 .cardPriceDetails .periodPlaceholder {
  font-weight: bold;
  padding: 0.1em;
}

.c1 .c3 {
  color: black;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="cardHeader"
          src="path to image"
        />
        <div
          class="c3 c4"
        >
          <div>
            title
          </div>
        </div>
      </div>
      <div
        class="c3 c5"
      >
        <div>
          subtitle
        </div>
        <br />
        <hr />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
